prow/pubsub/subscriber/subscriber.go

```diff
@@ -154,16 +172,70 @@ func (peh *periodicJobHandler) getProwJobSpec(cfg *config.Config, pe ProwJobEven
 }
 
 // presubmitJobHandler implements jobHandler
-type presubmitJobHandler struct{}
+type presubmitJobHandler struct {
+	GitHubClient githubClient
+	GitClient    git.ClientFactory
+}
+
+func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, error) {
+	// presubmit jobs require Refs and Refs.Pulls to be set
+	refs := pe.Refs
+	if refs == nil {
+		return nil, nil, errors.New("Refs must be supplied")
+	}
+	if len(refs.Pulls) != 1 {
+		return nil, nil, fmt.Errorf("expect 1 pull, got: %d", len(refs.Pulls))
+	}
+	pr := refs.Pulls[0]
+
+	var presubmitJob *config.Presubmit
+	org, repo, branch := refs.Org, refs.Repo, pr.Ref
+	orgRepo := org + "/" + repo
+	baseSHAGetter := func() (string, error) {
+		if refs.BaseSHA != "" {
+			return refs.BaseSHA, nil
+		}
+		// Fall back to fetch from GitHub
+		baseSHA, err := prh.GitHubClient.GetRef(refs.Org, refs.Repo, "heads/"+refs.BaseRef)
+		if err != nil {
+			return "", fmt.Errorf("failed to get baseSHA: %v", err)
+		}
+		return baseSHA, nil
+	}
+	headSHAGetter := func() (string, error) {
+		return pr.SHA, nil
+	}
+
+	// This will work with inrepoconfig
+	presubmits, err := cfg.GetPresubmits(prh.GitClient, orgRepo, baseSHAGetter, headSHAGetter)
+	if err != nil {
+		// Fall back to static presubmits to avoid deadlocking when a presubmit is used to verify
+		// inrepoconfig
+		logrus.WithError(err).Debug("Failed to get presubmits")
+		presubmits = cfg.GetPresubmitsStatic(orgRepo)
+	}
+
+	for _, job := range presubmits {
+		if !job.CouldRun(branch) { // filter out jobs that are not branch matching
+			continue
+		}
+		if job.Name == pe.Name {
+			presubmitJob = &job // running multiple jobs with the same name for a PR is not a good idea, so take the first one
```

Comment:
If there are multiple jobs with the same name that match this could be very confusing to debug, so it would probably be a good idea to at least log a warning that multiple jobs matched, or maybe even fail to trigger the jobs all together and return an error..

Reply:
returns an error instead